src/cfml/system/services/ServerService.cfc

```diff
@@ -424,10 +483,11 @@ component accessors="true" singleton {
 		if( !arguments.all ){
 			var servers 	= getServers();
 			var serverdir 	= getCustomServerFolder( arguments.serverInfo );
-			var serverJSONPath = arguments.serverInfo.webroot & '/server.json';
+			var defaultServerJSONPath = arguments.serverInfo.webroot & '/server.json';
+			var serverJSONPath = arguments.serverInfo.webroot & '/server-#arguments.serverInfo.name#.json';
 
 			// try to delete from config first
-			structDelete( servers, hash( arguments.serverInfo.webroot ) );
+			structDelete( servers, hash( arguments.serverInfo.webroot & arguments.serverInfo.name ) );
```

Comment:
Hash needs improved. Possibly to include the web root, name AND version of the server.